src/main/java/com/simple/metrics/kafka/DropwizardReporter.java

```diff
@@ -61,10 +65,14 @@ public void metricRemoval(KafkaMetric kafkaMetric) {
         String name = dropwizardMetricName(kafkaMetric);
         LOGGER.debug("Removing {}", name);
         registry.remove(name);
+        metricNames.remove(name);
```

Comment:
Looks like [HashSet.remove](https://docs.oracle.com/javase/7/docs/api/java/util/HashSet.html#remove(java.lang.Object)) is safe to call on a metric that's already been removed, so this should be fine.